controllers/sample_controller_rendered_resources.go

```diff
@@ -0,0 +1,359 @@
+
+	if err := r.ssaStatus(ctx, objectInstance); err != nil {
+		r.Event(objectInstance, "Warning", "ErrorUpdatingStatus", fmt.Sprintf("updating state to %v", string(status.State)))
+		return fmt.Errorf("error while updating status %s to: %w", status.State, err)
+	}
+
+	r.Event(objectInstance, "Normal", "StatusUpdated", fmt.Sprintf("updating state to %v", string(status.State)))
+	return nil
+}
+
+func getStatusFromObjectInstance(objectInstance *v1alpha1.Sample) v1alpha1.SampleStatus {
+	return objectInstance.Status
+}
+
+// getResourcesFromLocalPath returns resources from the dirPath in unstructured format.
+// Only one file in .yaml or .yml format should be present in the target directory.
+func getResourcesFromLocalPath(dirPath string, logger logr.Logger) (*ManifestResources, error) {
+	dirEntries := make([]fs.DirEntry, 0)
+	err := filepath.WalkDir(dirPath, func(path string, info fs.DirEntry, err error) error {
+		// initial error
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			return nil
+		}
+		dirEntries, err = os.ReadDir(dirPath)
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	childCount := len(dirEntries)
+	if childCount == 0 {
+		logger.V(debugLogLevel).Info(fmt.Sprintf("no yaml file found at file path %s", dirPath))
+		return nil, nil
+	} else if childCount > 1 {
+		logger.V(debugLogLevel).Info(fmt.Sprintf("more than one yaml file found at file path %s", dirPath))
+		return nil, nil
+	}
+	file := dirEntries[0]
+	allowedExtns := sets.NewString(".yaml", ".yml")
+	if !allowedExtns.Has(filepath.Ext(file.Name())) {
+		return nil, nil
+	}
+
+	fileBytes, err := os.ReadFile(filepath.Join(dirPath, file.Name()))
+	if err != nil {
+		return nil, fmt.Errorf("yaml file could not be read %s in dir %s: %w", file.Name(), dirPath, err)
+	}
+	return parseManifestStringToObjects(string(fileBytes))
+}
+
+// parseManifestStringToObjects parses the string of resources into a list of unstructured resources.
+func parseManifestStringToObjects(manifest string) (*ManifestResources, error) {
+	objects := &ManifestResources{}
+	reader := yamlUtil.NewYAMLReader(bufio.NewReader(strings.NewReader(manifest)))
+	for {
+		rawBytes, err := reader.Read()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				return objects, nil
+			}
+
+			return nil, fmt.Errorf("invalid YAML doc: %w", err)
+		}
+
+		rawBytes = bytes.TrimSpace(rawBytes)
+		unstructuredObj := unstructured.Unstructured{}
+		if err := yaml.Unmarshal(rawBytes, &unstructuredObj); err != nil {
+			objects.Blobs = append(objects.Blobs, append(bytes.TrimPrefix(rawBytes, []byte("---\n")), '\n'))
+		}
+
+		if len(rawBytes) == 0 || bytes.Equal(rawBytes, []byte("null")) || len(unstructuredObj.Object) == 0 {
+			continue
+		}
+
+		objects.Items = append(objects.Items, &unstructuredObj)
+	}
+}
+
+// ssaStatus patches status using SSA on the passed object
+func (r *SampleReconciler) ssaStatus(ctx context.Context, obj client.Object) error {
+	obj.SetManagedFields(nil)
```

Comment:
I think you should remove the resource version to avoid optimistic locking

Reply:
I will add it just to be sure, but in this simple use case it shouldn't matter